doc/nrf/protocols/matter/end_product/versioning.rst

```diff
@@ -0,0 +1,88 @@
+.. _ug_versioning_in_matter:
+
+Maintaining firmware version
+############################
+
+.. contents::
+   :local:
+   :depth: 2
+
+To effectively deploy a Matter product, it is essential to implement application version management. This versioning is crucial for managing firmware upgrades on devices, and it is also displayed within ecosystem applications, as provided by the Basic Information cluster.
+
+There are two primary approaches for maintaining versioning. It's important to note that these approaches should not be used simultaneously:
+
+Utilizing a :file:`VERSION` file as detailed in the Zephyr Project documentation: :ref:`Zephyr Project Versioning<app-version-details>`. This method involves defining the version information in a specific file format.
+
+Implementing dedicated Kconfig configurations. This approach uses Kconfig system configurations to set and manage the versioning details.
+
+Choose the approach that best aligns with your project requirements and infrastructure.
+
+Using :file:`VERSION` file
+**************************
+
+To implement versioning based on a :file:`VERSION` file, you must create a file named :file:`VERSION` within the application directory and ensure it contains the appropriate content:
+
+.. code-block:: console
+
+    VERSION_MAJOR =
+    VERSION_MINOR =
+    PATCHLEVEL =
+    VERSION_TWEAK =
+
+.. note::
+   You must assign a value to at least one of the variables.
+   If not, the :file:`VERSION` file will cause error.
+
+For example:
+
+.. code-block:: console
+
+   VERSION_MAJOR = 2
+   VERSION_MINOR = 5
+   PATCHLEVEL = 99
+   VERSION_TWEAK = 0
+
+A :file:`VERSION` file is responsible for assigning values to MCUboot version in the following format: ``MAJOR.MINOR.PATCH.TWEAK``.
+
+Using KConfig options
+*********************
+
+Depending on how you transfer the updated images to the device, you need to use different Kconfig options.
+
+.. _ug_matter_dfu_ota:
+
+Device Firmware Upgrade over Matter versioning
+==============================================
+
+When using DFU over Matter, you must edit the :file:`prj.conf` file in the application directory.
+Add the following Kconfig options to change the version:
+
+* :kconfig:option:`CONFIG_CHIP_DEVICE_SOFTWARE_VERSION` to set to the version number.
+* :kconfig:option:`CONFIG_CHIP_DEVICE_SOFTWARE_VERSION_STRING` to set the version string.
+
+Additionally, since Nordic chips use MCUboot Image Tool, you need to also edit the :kconfig:option:`CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION` Kconfig option, with a value in the following format: ``"MAJOR . MINOR . PATCHLEVEL . TWEAK"``.
```

Comment:
One more comment.. Here i do see " . (dot) TWEAK" but in below examples, i do see "+" sign. Maybe all is fine, but it might be confusing.

Reply:
Good point, I didn't catch that